daemon/cmd/agenthealth.go

```diff
@@ -46,14 +47,17 @@ func setsockoptReuseAddrAndPort(network, address string, c syscall.RawConn) erro
 	return soerr
 }
 
-// startAgentHealthHTTPService registers a handler function for the /healthz
-// status HTTP endpoint exposed on addr. This endpoint reports the agent health
-// status and is equivalent to what the `cilium status --brief` CLI tool reports.
-func (d *Daemon) startAgentHealthHTTPService(addr string) {
-	lc := net.ListenConfig{Control: setsockoptReuseAddrAndPort}
-	ln, err := lc.Listen(context.Background(), "tcp", addr)
-	if err != nil {
-		log.WithError(err).Fatalf("Unable to listen on %s for healthz status API server", addr)
+// startAgentHealthHTTPService registers a handler function for the /healthz status HTTP endpoint
+// exposed on localhost (127.0.0.1 and/or ::1, depending on the values of ipv4 and ipv6). This
+// endpoint reports the agent health status and is equivalent to what the `cilium status --brief`
+// CLI tool reports.
+func (d *Daemon) startAgentHealthHTTPService(port int, ipv4, ipv6 bool) {
```

Comment:
nit: Maybe we can import and use `options` directly instead of passing options as arguments?

Reply:
The idea was to decouple the `(d *Daemon) startAgentHealthHTTPService` from from `pkg/options` in order to make it easier to re-factor and re-use in the future. But I guess this is so tightly coupled to the daemon anyway that it doesn't matter. Changed to use `pkg/option` and also take `AgentHealthPort` from there.